compiler/src/main/java/org/qbicc/graph/atomic/AtomicTraits.java

```diff
@@ -0,0 +1,25 @@
+package org.qbicc.graph.atomic;
+
+/**
+ * A set of bits used to represent the set of capabilities that an atomic operation may have, in order to determine which
+ * operations include which other operations.
+ */
```

Comment:
I implemented this all in one place so that it's easier to analyze and debug problems.